system/mp/test/mptest.cpp

```diff
@@ -1557,9 +1600,12 @@ int main(int argc, char* argv[])
     {
         EnableSEHtoExceptionMapping();
         StringBuffer lf;
-
+        startMPI();
```

Comment:
We needed this to get the global rank so that we can workout the port for each process. Technically you can run the mp tests the same way as mpi test by using mpirun. If we are completely eliminating this for tests I can do something about this.

Reply:
We don't want to have to run via mpirun , if not testing MPI.

>We needed this to get the global rank so that we can workout the port for each process. 

Unless started via MPI and therefore can get global rank, then it should be using the port as provided on the command line.

Reply:
MPI is always there whether or not if you start the process with "mpirun". If "mpirun" wasn't used, the global rank will be 0 for everyone, so whatever the port provided by the commandline will be the port. I think easiest thing we can do is to move this after figuring out "-mpi" or "-mp". Then call startMPI/stopMPI/getGlobalRank based on that. wdyt?

Reply:
>MPI is always there whether or not if you start the process with "mpirun"

yes, that's what I meant. It shouldn't be.

>I think easiest thing we can do is to move this after figuring out "-mpi" or "-mp". Then call startMPI/stopMPI/getGlobalRank based on that. wdyt?

yes agree.